File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.java
class Solution {
    public int[] replaceElements(int[] arr) {

         int max_right = -1;

        for(int i = arr.length -1; i>=0; i--){
            int  temp = arr[i];
            arr[i] = max_right;
            if(temp > max_right){
                max_right = temp;
            }
        }
        return arr;
    }
}

File: Data Structures & Algorithms/kth-distinct-string-in-an-array/submission-0.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:

        freq = {}

        # Count frequency
        for string in arr:
            freq[string] = freq.get(string, 0) + 1

        # Find k-th distinct string
        for string in freq:
            if freq[string] == 1:
                k -= 1

                if k == 0:
                    return string

        return ""
        